src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/library.cljs

```diff
@@ -78,76 +92,103 @@
        :ok-button #(react/call :on-ok this)}])
    :form
    (fn [{:keys [props state]}]
-     (let [library-schema (:library-schema props)]
+     (let [library-schema (:library-schema props)
+           existing (get-in props [:workspace :workspace :library-attributes])]
        [:div {:style {:width "50vw"}}
+        (when (:saving? @state)
+          [comps/Blocker {:banner "Submitting library attributes..."}])
         [:div {:style {:maxHeight 400 :overflowY "auto"
                        :padding "0.5em" :border style/standard-line
                        :marginBottom "1em"}}
          (map (fn [property-key]
-                (let [{:keys [hidden title inputHint enum]} (get-in library-schema [:properties property-key])
+                (let [{:keys [hidden title inputHint enum type items minimum default]} (get-in library-schema [:properties property-key])
                       required? (contains? (:required-props @state) property-key)
                       pk-str (name property-key)]
                   (when-not hidden
                     [:div {}
-                     (style/create-form-label (str title (when required? " (required)")))
-                     (cond enum (style/create-identity-select {:ref pk-str} enum)
-                           required? [input/TextField {:ref pk-str
-                                                       :style {:width "100%"}
-                                                       :placeholder inputHint
-                                                       :predicates [(when required? (input/nonempty pk-str))]}]
-                           :else (style/create-text-field {:ref (name property-key)
-                                                           :style {:width "100%"}
-                                                           :placeholder inputHint}))])))
+                     (style/create-form-label
+                       [:div {:style {:display "flex" :alignItems "baseline"}}
+                        [:div {:style {:flex "0 0 auto"}}
+                         title
+                         (when required?
+                           [:b {:style {:marginLeft "1ex"}} "(required)"])]
+                        [:div {:style {:flex "1 1 auto"}}]
+                        (when-not enum
+                          [:i {:style {:flex "0 0 auto"}}
+                           (if (= type "array")
+                             (str "Comma-separated list of " (:type items) "s")
+                             (clojure.string/capitalize type))])])
+                     (if enum
+                       (style/create-identity-select {:ref pk-str
+                                                      :defaultValue (get existing property-key ENUM_EMPTY_CHOICE)}
+                                                     (cons ENUM_EMPTY_CHOICE enum))
```

Comment:
Very strange.  I don't want to do that `disabled selected value` thing because I want people to be able to go back to "no selection".  Maybe a better label for the empty option?